ibm/resource_ibm_resource_key.go

```diff
@@ -184,28 +263,32 @@ func resourceIBMResourceKeyUpdate(d *schema.ResourceData, meta interface{}) erro
 }
 
 func resourceIBMResourceKeyRead(d *schema.ResourceData, meta interface{}) error {
-	rsContClient, err := meta.(ClientSession).ResourceControllerAPI()
+	rsContClient, err := meta.(ClientSession).ResourceControllerV2API()
 	if err != nil {
 		return err
 	}
 	resourceKeyID := d.Id()
+	resourceKeyGet := rc.GetResourceKeyOptions{
+		ID: &resourceKeyID,
+	}
 
-	resourceKey, err := rsContClient.ResourceServiceKey().GetKey(resourceKeyID)
+	resourceKey, resp, err := rsContClient.GetResourceKey(&resourceKeyGet)
 	if err != nil {
-		return fmt.Errorf("Error retrieving resource key: %s", err)
+		return fmt.Errorf("Error retrieving resource key: %s with resp code: %s", err, resp)
 	}
-	d.Set("credentials", Flatten(resourceKey.Credentials))
+	d.Set("credentials", resourceKey.Credentials)
```

Comment:
It would break existing customer..Schema is of type map[string]interface{}
have to conver to map[string]interface{} and set.. something like..
```

var credInterface map[string]interface{}
cred, _ := json.Marshal(resourceKey.Credentials)
json.Unmarshal(cred, &credInterface)
d.Set("credentials", Flatten(credInterface))
```

Reply:
done